libzecale/circuits/pairing/bls12_377_pairing.hpp

```diff
@@ -6,33 +6,52 @@
 #define __ZECALE_CIRCUITS_PAIRING_BLS12_377_PAIRING_HPP__
 
 #include "libzecale/circuits/fields/fp12_2over3over2_gadgets.hpp"
+#include "libzecale/circuits/pairing/bw6_761_pairing_params.hpp"
+#include "libzecale/circuits/pairing/pairing_params.hpp"
 
 #include <libff/algebra/curves/bls12_377/bls12_377_pp.hpp>
+#include <libsnark/gadgetlib1/gadgets/curves/weierstrass_g1_gadget.hpp>
+#include <libsnark/gadgetlib1/gadgets/curves/weierstrass_g2_gadget.hpp>
 #include <libsnark/gadgetlib1/gadgets/fields/fp2_gadgets.hpp>
-#include <libsnark/gadgetlib1/gadgets/pairing/pairing_params.hpp>
 
 namespace libzecale
 {
 
+template<typename ppT> class bls12_377_G1_precomputation
+{
+public:
+    using FieldT = libff::Fr<ppT>;
+
+    std::shared_ptr<libsnark::pb_linear_combination<FieldT>> _Px;
+    std::shared_ptr<libsnark::pb_linear_combination<FieldT>> _Py;
+
+    bls12_377_G1_precomputation();
+
+    // Construct a populated G1_precomputation from a value. All terms are
+    // created as constants, generating no cost in the circuit.
```

Comment:
`generating no cost in the circuit`, what do you mean here? Do you mean "this function does not add any gates in the circuit", hence "doesn't alter the prover perfs"?

Reply:
reworded the comment